zapcore/json_encoder.go

```diff
@@ -60,10 +52,10 @@ type jsonEncoder struct {
 // libraries will ignore duplicate key-value pairs (typically keeping the last
 // pair) when unmarshaling, but users should attempt to avoid adding duplicate
 // keys.
-func NewJSONEncoder(cfg JSONConfig) Encoder {
+func NewJSONEncoder(cfg EncoderConfig) Encoder {
```

Comment:
Accepting a pointer to config suggests, to me, that the config is optional. In this case, it's not.

In the case when we have both JSON and console encoders running, I don't think they'll share configs - they'll want different time and level formatters, at least.